flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -149,7 +149,9 @@ public void run() {
                     // we can close this consumer thread once we've reached the end of the
                     // subscribed shard
                     break;
-                } else if (result == CANCELLED) {
+                } else if (result == CANCELLED && isRunning()) {
+                    // cancellation might be signalled as part of graceful shutdown during
+                    // stop-with-savepoint, we should ignore it when the consumer is not running
```

Comment:
nit: Since we have changed the `shutdownFetcher` method, I think `running == false` is going to be the new normal shutdown shutdown procedure for EFO

Reply:
Removed the notion of stop-with-savepoint